### ISSUE TITLE
luabitop won't build on osx

### ISSUE THREAD

**Issue by maun** [NixOS/nixpkgs#6260]
CC is set to gcc which is not available. Changing it leads to linker errors

```
clang -shared -fPIC  -o bit.so bit.o
Undefined symbols for architecture x86_64:
  "_luaL_checknumber", referenced from:
      _luaopen_bit in bit.o
      _bit_tobit in bit.o
      _bit_bnot in bit.o
      _bit_band in bit.o
      _bit_bor in bit.o
      _bit_bxor in bit.o
      _bit_lshift in bit.o
      ...
  "_luaL_error", referenced from:
      _luaopen_bit in bit.o
  "_luaL_setfuncs", referenced from:
      _luaopen_bit in bit.o
  "_lua_createtable", referenced from:
      _luaopen_bit in bit.o
  "_lua_gettop", referenced from:
      _bit_band in bit.o
      _bit_bor in bit.o
      _bit_bxor in bit.o
  "_lua_pushlstring", referenced from:
      _bit_tohex in bit.o
  "_lua_pushnumber", referenced from:
      _luaopen_bit in bit.o
      _bit_tobit in bit.o
      _bit_bnot in bit.o
      _bit_band in bit.o
      _bit_bor in bit.o
      _bit_bxor in bit.o
      _bit_lshift in bit.o
      ...
  "_lua_type", referenced from:
      _bit_tohex in bit.o
ld: symbol(s) not found for architecture x86_64
clang-3.5: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Profpatsch** [comment]
(triage) status?

**Comment by c0bw3b** [comment]
Fixed in #8261
